src/gocert.py

```diff
@@ -113,3 +131,70 @@ def create_first_user(self, username: str, password: str) -> int | None:
         logger.info("created the first user in GoCert.")
         id = req.json().get("id")
         return int(id) if id else None
+
+    def get_certificate_requests_table(self, token: str) -> CertificateRequests | None:
+        """Get all certificate requests table from GoCert."""
+        try:
+            res = requests.get(
+                f"{self.url}/api/{self.API_VERSION}/certificate_requests",
+                verify=self.ca_path,
+                headers={"Authorization": f"Bearer {token}"},
+            )
+            res.raise_for_status()
+        except requests.RequestException as e:
+            logger.warning(
+                "couldn't retrieve certificate requests table: code %s, %s",
+                e.response.status_code if e.response else "unknown",
+                e.response.text if e.response else "unknown",
+            )
+            return None
+        except OSError:
+            logger.warning("error occurred during HTTP request: TLS file invalid")
+            return None
+        table = res.json()
+        return CertificateRequests(
+            rows=[
+                CertificateRequest(
+                    row.get("id"),
+                    row.get("csr"),
+                    validate(row.get("certificate")),
+                )
+                for row in table
+            ]
+            if table
+            else []
+        )
+
+    def post_csr(self, csr: str, token: str) -> None:
+        """Post a new CSR to GoCert."""
+        try:
+            res = requests.post(
+                f"{self.url}/api/{self.API_VERSION}/certificate_requests",
+                verify=self.ca_path,
+                headers={"Authorization": f"Bearer {token}"},
+                data=csr,
+            )
+            res.raise_for_status()
+        except requests.RequestException as e:
+            logger.error(
+                "couldn't post new certificate requests: code %s, %s",
+                e.response.status_code if e.response else "unknown",
+                e.response.text if e.response else "unknown",
+            )
+        except OSError:
+            logger.warning("error occurred during HTTP request: TLS file invalid")
+
+
+def validate(pem_string: str) -> list[str] | Literal["", "rejected"]:
```

Comment:
Don't we do such validation on the GoCert level too? I'm fine double checking, just asking.

Reply:
yes, i think validation is a wrong word, this is more like formatting the string response. I'm gonna call it serialize.